recipes/sensor.rb

```diff
@@ -0,0 +1,61 @@
+# install and run the metaflows sensor/collector
+include_recipe 'runit'
+
+if node['platform'] == 'centos'
+  package 'expect'
+  package 'expectk'
+
+  cookbook_file "/etc/chef/register_sensor" do
+    source 'register_sensor'
+    owner 'metaflows'
+    group 'metaflows'
+    mode '0755'
+    action :create_if_missing
+  end
+
+  socrata_configuration_kms "s3://#{node['metaflows']['clortho_bucket']}/#{node['metaflows']['metaflows_clortho_file']}" do
+    local_path "/dev/shm/#{node['metaflows']['metaflows_clortho_file']}"
+    owner "metaflows"
+    group "metaflows"
+    mode 0600
+    region node['cloud']['region']
+    notifies :run, 'bash[register sensor]', :immediately
+  end
+
+  sensor_status = File.open("/nsm/etc/UUID", &:readline)
+
+  # overwrite the tty requirement in CentOS so we can run scripts with scripts and sudo
+  # an expect script will answer the prompt and register the sensor
+  bash 'register sensor' do
+    code <<-EOC
+    set -e
+    sed -i '/Defaults    requiretty/c\# Defaults    requiretty' /etc/sudoers
```

Comment:
Normally I get antsy around shelling out to run `sed`. This would conflict with the `sudo` cookbook if both were ever run on the same node--it and these commands would both be modifying the sudoers file on every Chef run, overwriting each other's changes.